sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -64,3 +65,40 @@ def _get_data_objects(
             )
             for row in df[["database_name", "schema_name", "name", "kind"]].itertuples()
         ]
+
+    def _insert_append_pandas_df(
+        self,
+        table_name: TableName,
+        df: pd.DataFrame,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+    ) -> None:
+        from snowflake.connector.pandas_tools import write_pandas
+
+        table = exp.to_table(table_name)
```

Comment:
this, *db = tuple(reverse(part.name if part.quoted part.name.upper() for part in table.parts))

maybe?

Reply:
nice!